src/main/java/ee/buerokratt/ruuter/domain/steps/ReturnStep.java

```diff
@@ -17,7 +18,10 @@ public class ReturnStep extends ConfigurationStep {
 
     @Override
     protected void executeStepAction(ConfigurationInstance ci) {
-        ci.setReturnValue(ci.getScriptingHelper().evaluateScripts(returnValue, ci.getContext(), ci.getRequestBody(), ci.getRequestParams()));
+        ci.setReturnValue((HttpStepResult) ci.getScriptingHelper().evaluateScripts(returnValue, ci.getContext(), ci.getRequestBody(), ci.getRequestParams()));
+        if (ci.getProperties().getFinalResponse() != null && ci.getProperties().getFinalResponse().getHttpStatusCode() != null) {
```

Comment:
In theory, if you instantiated the "FinalResponse" object in ApplicationProperties, then you dont need to check if its `null` 

If thats the case, you could also extract `ci.getProperties().getFinalResponse().getHttpStatusCode()` into a variable